### PR TITLE
[CWS] Add spans on fargate

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

This PR:
* adds the support of retrieving APM spans on fargate.
* updates the exec span functional test to use a thread (as for open)

### Motivation

Leverage APM tracing on fargate

### Additional Notes

For now only java is supported on APM side

### Possible Drawbacks / Trade-offs

If a TLS memory was registered it will add a little overhead, otherwise not.

### Describe how to test/QA your changes

You can play with the syscall tester as follow:
```
sudo ./bin/cws-instrumentation/cws-instrumentation trace --probe-addr= --disable-proc-scan --verbose -- ./pkg/security/tests/syscall_tester/bin/syscall_tester span-open 42 51 foobar
sudo ./bin/cws-instrumentation/cws-instrumentation trace --probe-addr= --disable-proc-scan --verbose -- ./pkg/security/tests/syscall_tester/bin/syscall_tester span-exec 42 51 /usr/bin/touch /tmp/foo
```

### CODE REVIEW

File: pkg/security/ebpf/c/include/hooks/exec.h [hunk 1]
```diff
@@ -726,6 +726,9 @@ int __attribute__((always_inline)) send_exec_event(ctx_t *ctx) {
     // send the entry to maintain userspace cache
     send_event_ptr(ctx, EVENT_EXEC, event);
 
+    // as previously registered memory will become unreachable, we'll have to unregister the TLS
+    unregister_span_memory();
+
```
Comment (Gui774ume) [inline pkg/security/ebpf/c/include/hooks/exec.h:731]: 👋🏻 Have you double checked we can still identify the original span for an RCE ? 🤔 
I think this was the original rational not to remove it here ...

Comment (Gui774ume) [reply pkg/security/ebpf/c/include/hooks/exec.h:731]: I _**think**_ it's ok though

Comment (paulcacheux) [reply pkg/security/ebpf/c/include/hooks/exec.h:731]: yeah let's do the ebpf at least in another PR, will be easier to revert if needed and will be easier to track as a potential change

Comment (spikat) [reply pkg/security/ebpf/c/include/hooks/exec.h:731]: Ok, let me split it into another one for this change

**Review by YoannGh — APPROVED** [review]

**Pull request merged**